app/javascript/gobierto_data/webapp/pages/Visualizations.vue

```diff
@@ -0,0 +1,160 @@
+          />
+        </div>
+      </template>
+    </div>
+  </div>
+</template>
+<script>
+import VisualizationsGrid from "./../components/landingviz/VisualizationsGrid";
+import { CategoriesMixin } from "./../../lib/mixins/categories.mixin";
+import { convertToCSV } from "./../../lib/helpers";
+import { FiltersMixin } from "./../../lib/mixins/filters.mixin";
+import { VisualizationFactoryMixin } from "./../../lib/factories/visualizations";
+import { DataFactoryMixin } from "./../../lib/factories/data";
+import { QueriesFactoryMixin } from "./../../lib/factories/queries";
+import { SkeletonSpinner } from "lib/vue-components";
+
+export default {
+  name: "Visualizations",
+  components: {
+    VisualizationsGrid,
+    SkeletonSpinner
+  },
+  mixins: [
+    CategoriesMixin,
+    FiltersMixin,
+    VisualizationFactoryMixin,
+    DataFactoryMixin,
+    QueriesFactoryMixin
+  ],
+  data() {
+    return {
+      datasetsArray: [],
+      datasetsVisualizations: [],
+      publicVisualizations: [],
+      isVizsLoaded: false,
+      labelVisualizations: I18n.t("gobierto_data.projects.visualizations") || ""
+    }
+  },
+  computed: {
+    showVizs() {
+      return this.datasetsArray.length && this.datasetsVisualizations.length
+    }
+  },
+  created() {
+    this.getDataVizs()
+  },
+  methods: {
+    async getDataVizs() {
+      const { data: response } = await this.getListVisualizations();
+      const { data } = response
+      let listVisualizations = data.slice(0, 4)
+      let datasets = listVisualizations.map(dataset => dataset.attributes.dataset_id);
+      datasets = [ ...new Set(datasets) ];
+      this.getDatasets(datasets)
+      this.datasetsArray = datasets
+    },
+    async getDatasets(datasetsID) {
+      //Filters dataset which only contains Visualizations
+      const datasetsWithVizs = []
+      for (let index = 0; index < datasetsID.length; index++) {
+        const { data: response } = await this.getVisualizations({
+          "filter[dataset_id]": datasetsID[index]
+        });
+        const { data } = response;
+
+        if (data.length) {
+          const { relationships: { dataset: { data: { id } } } } = data[0];
+          datasetsWithVizs.push(id);
+        }
+      }
+      this.datasetsVisualizations = this.subsetItems.filter((dataset) => datasetsWithVizs.includes(dataset.id))
+      this.getPublicVisualizations()
+    },
+    async getPublicVisualizations() {
+      let allVizs = []
+      for (let index = 0; index < this.datasetsArray.length; index++) {
+        const { data: response } = await this.getVisualizations({
+          "filter[dataset_id]": this.datasetsArray[index]
+        });
+        const { data } = response;
+
+        if (data.length) {
+          this.publicVisualizations = await this.getDataFromVisualizations(data);
+          allVizs.push(this.publicVisualizations);
+        }
+      }
+
+      this.publicVisualizations = allVizs.reduce((flatten, arr) => [...flatten, ...arr])
```

Comment:
If you're trying to flat an array, ES6 `Array` class already provides an method for it
```suggestion
      this.publicVisualizations = allVizs.flat()
```